aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -148,6 +159,39 @@ private void associateKmsKey(final AmazonWebServicesClientProxy proxy,
         logger.log(kmsKeyMessage);
     }
 
+    private void updateTags(final AmazonWebServicesClientProxy proxy,
+                            final ResourceModel previousModel,
+                            final ResourceModel model,
+                            final Logger logger) {
+        final Set<Tag> previousTags = Optional.ofNullable(previousModel).map(ResourceModel::getTags).orElse(new HashSet<>());
+        final Set<Tag> newTags = Optional.ofNullable(model.getTags()).orElse(new HashSet<>());
+        final Set<Tag> tagsToRemove = Sets.difference(previousTags, newTags);
+        final Set<Tag> tagsToAdd = Sets.difference(newTags, previousTags);
+        try {
+            if (!tagsToRemove.isEmpty()) {
+                final List<String> tagKeys = tagsToRemove.stream().map(Tag::getKey).collect(Collectors.toList());
+                proxy.injectCredentialsAndInvokeV2(Translator.translateToUntagLogGroupRequest(model.getLogGroupName(), tagKeys),
+                        ClientBuilder.getClient()::untagLogGroup);
+
+                final String message =
+                        String.format("%s [%s] successfully removed tags: [%s]",
+                                ResourceModel.TYPE_NAME, model.getLogGroupName(), tagKeys);
+                logger.log(message);
+            }
+            if(!tagsToAdd.isEmpty()) {
+                proxy.injectCredentialsAndInvokeV2(Translator.translateToTagLogGroupRequest(model.getLogGroupName(), tagsToAdd),
+                        ClientBuilder.getClient()::tagLogGroup);
+
+                final String message =
+                        String.format("%s [%s] successfully added tags: [%s]",
+                                ResourceModel.TYPE_NAME, model.getLogGroupName(), tagsToAdd);
+                logger.log(message);
+            }
+        } catch (final ResourceNotFoundException e) {
```

Comment:
https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/pull/306 should handle `AwsServiceExceptions` automatically but [the contract seems to still require some to be explicitly mapped](https://github.com/aws-cloudformation/cloudformation-cli/issues/527)

Reply:
I added one more exception handling, because the link provided is actually the wrong one, it is for CloudWatch not CloudWatch Logs. That's the correct one 😄 
https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_TagLogGroup.html#API_TagLogGroup_Errors